plugins/optimization-detective/optimization.php

```diff
@@ -80,14 +80,38 @@ static function ( string $output, ?int $phase ): string {
  * @access private
  */
 function od_maybe_add_template_output_buffer_filter(): void {
-	if (
-		! od_can_optimize_response() ||
-		od_is_rest_api_unavailable() ||
-		isset( $_GET['optimization_detective_disabled'] ) // phpcs:ignore WordPress.Security.NonceVerification.Recommended
-
-	) {
+	$conditions = array(
+		array(
+			'test'   => od_can_optimize_response(),
+			'reason' => __( 'Page is not optimized because od_can_optimize_response() returned false. This can be overridden with the od_can_optimize_response filter.', 'optimization-detective' ),
+		),
+		array(
+			'test'   => ! od_is_rest_api_unavailable() || ( wp_get_environment_type() === 'local' && ! class_exists( 'Test_OD_Optimization' ) ),
```

Comment:
@felixarntz:

> Not a blocker, but is there a more generic way to check if tests are being run? Maybe a constant that the WP core unit test suite always sets?

I suppose amore generic way would be to see if the `tests_add_filter()` function exists:

```suggestion
			'test'   => ! od_is_rest_api_unavailable() || ( wp_get_environment_type() === 'local' && ! function_exists( 'tests_add_filter' ) ),
```

Reply:
I think that would be a bit better.